storage/types.go

```diff
@@ -70,28 +75,33 @@ func bytesForBits(numBits int) int {
 func NewNodeIDFromHash(h []byte) NodeID {
 	return NodeID{
 		Path:          h,
-		PathLenBits:   len(h) * 8,
 		PrefixLenBits: len(h) * 8,
 	}
 }
 
 // NewEmptyNodeID creates a new zero-length NodeID with sufficient underlying
 // capacity to store a maximum of maxLenBits.
 func NewEmptyNodeID(maxLenBits int) NodeID {
+	if got, want := maxLenBits%8, 0; got != want {
+		panic(fmt.Sprintf("storeage: NewEmptyNodeID() maxLenBits mod 8: %v, want %v", got, want))
+	}
 	return NodeID{
-		Path:          make([]byte, bytesForBits(maxLenBits)),
+		Path:          make([]byte, maxLenBits/8),
 		PrefixLenBits: 0,
-		PathLenBits:   maxLenBits,
 	}
 }
 
 // NewNodeIDWithPrefix creates a new NodeID of nodeIDLen bits with the prefixLen MSBs set to prefix.
+// NewNodeIDWithPrefix places the lower prefixLenBits of prefix in the most significant bits of path.
+// Path will have enough bytes to hold maxLenBits
+//
 func NewNodeIDWithPrefix(prefix uint64, prefixLenBits, nodeIDLenBits, maxLenBits int) NodeID {
-	maxLenBytes := bytesForBits(maxLenBits)
+	if got, want := nodeIDLenBits%8, 0; got != want {
```

Comment:
Again I don't think this is an error, it's just something we don't currently do.